cmd/zpool/zpool_main.c

```diff
@@ -7252,16 +7252,69 @@ print_error_log(zpool_handle_t *zhp)
 	pathname = safe_malloc(len);
 	elem = NULL;
 	while ((elem = nvlist_next_nvpair(nverrlist, elem)) != NULL) {
-		nvlist_t *nv;
-		uint64_t dsobj, obj;
-
-		verify(nvpair_value_nvlist(elem, &nv) == 0);
-		verify(nvlist_lookup_uint64(nv, ZPOOL_ERR_DATASET,
-		    &dsobj) == 0);
-		verify(nvlist_lookup_uint64(nv, ZPOOL_ERR_OBJECT,
-		    &obj) == 0);
+		unsigned int error_count_blkids, error_count_levels;
+		uint64_t data_block_size, indirect_block_size;
+
+		nvlist_t *object_nv = fnvpair_value_nvlist(elem);
+		uint64_t dsobj = fnvlist_lookup_uint64(object_nv,
+		    ZPOOL_ERR_DATASET);
+		uint64_t obj = fnvlist_lookup_uint64(object_nv,
+		    ZPOOL_ERR_OBJECT);
+		uint64_t *block_ids = fnvlist_lookup_uint64_array(object_nv,
+		    ZPOOL_ERR_BLOCKID, &error_count_blkids);
+		int64_t *indirect_levels = fnvlist_lookup_int64_array(object_nv,
```

Comment:
Same here, this can be moved to inside of `if (zpool_get_block_size() == 0) {  }` block